design/baremetal-operator/bmh_non-bootable_iso.md

```diff
@@ -0,0 +1,73 @@
+# Add non-bootable iso attach API to BareMetalHost
+
+## Status
+
+Implementable
+
+## Summary
+
+Add a new field in BareMetalHost so that it is possible to attach a generic
+non-bootable iso image via Ironic, after the host has been provisioned.
+
+## Motivation
+
+In certain scenarios it is desirable to attach a non-bootable iso image
+to a system after provisioning has finished, for example :
+
+As a config data disk to do some post provisioning configuration for the
+system.
+
+For troubleshooting of the node.
+
+### Goals
+
+Expose the Ironic API to attach non-bootable iso images via Metal3.
+
+## Proposal
+
+### BMO Proposal
+
+Add a new optional section in the BareMetalHost spec containing details of the
+non-bootable iso image, example :
```

Comment:
Do we want to support non-ISO images, e.g. a USB stick? If not, please put to the non-goals section (which I don't see here but it's in the template).